03/app.js

```diff
@@ -5,17 +5,72 @@ import Category from './Category';
 import Cart from './Cart';
 
 import data from './data.json';
+import Product from './Product';
 
 class App extends React.Component {
     state = {
         cart: [],
     }
-    
+
+    addProductToCart = (product) => {
```

Comment:
Niestety to rozwiązanie nie działa prawidłowo.
Dodaj do koszyka wszystkie produkty, a potem usuń jeden z nich.
Niestety cały koszyk się "resetuje". Spróbuj to poprawić.